src/ui/components/shared/UserSettingsModal/PreferencesSettings.tsx

```diff
@@ -0,0 +1,178 @@
+
+import { updateEnableRepaint } from "protocol/enable-repaint";
+import { handleIntercomLogout } from "ui/utils/intercom";
+import useAuth0 from "ui/utils/useAuth0";
+import hooks from "ui/hooks";
+import * as actions from "ui/actions/app";
+import * as selectors from "ui/reducers/app";
+import { UIState } from "ui/state";
+import { SettingsTabTitle } from "ui/state/app";
+import { ApiKey, UserSettings } from "ui/types";
+
+import APIKeys from "../APIKeys";
+import SettingsModal from "../SettingsModal";
+import { Settings } from "../SettingsModal/types";
+import { SettingsBodyHeader } from "../SettingsModal/SettingsBody";
+
+import { getFeatureFlag } from "ui/utils/launchdarkly";
+import { AvatarImage } from "ui/components/Avatar";
+import { prefs } from "ui/utils/prefs";
+import Checkbox from "../Forms/Checkbox";
+import { disableEmailPreference, EmailPreference, enableEmailPreference } from "ui/hooks/users";
+import Radio from "../Forms/Radio";
+
+const EMAIL_NOTIFICATIONS = {
+  [EmailPreference.COLLABORATOR_REQUESTS]: "When somebody invites you to collaborate on a replay",
+  [EmailPreference.REPLAY_COMMENTS]: "When somebody comments on your replay",
+  [EmailPreference.NEW_TEAM_INVITE]: "When you're invited to a new team",
+};
+
+function Notification({
+  content,
+  preference,
+  emailPreferences,
+}: {
+  content: string;
+  preference: EmailPreference;
+  emailPreferences: EmailPreference[];
+}) {
+  const updateUserEmailPreferences = hooks.useUpdateUserEmailPreferences();
+
+  // The preferences list is a blacklist of preferences that are disabled.
+  const checked = !emailPreferences.includes(preference);
+
+  const handleChange = () => {
+    let newPrefs;
+    if (checked) {
+      newPrefs = disableEmailPreference(preference, emailPreferences);
+    } else {
+      newPrefs = enableEmailPreference(preference, emailPreferences);
+    }
+
+    updateUserEmailPreferences({ variables: { emailPreferences: newPrefs } });
+  };
+
+  return (
+    <div className="flex items-center space-x-2 cursor-pointer" data-private>
+      <Checkbox id={preference} checked={checked} onChange={handleChange} />
+      <label className="cursor-pointer" htmlFor={preference}>
+        <div>{content}</div>
+      </label>
+    </div>
+  );
+}
+
+function MarketingPreferences({ emailPreferences }: { emailPreferences: EmailPreference[] }) {
+  const updateUserEmailPreferences = hooks.useUpdateUserEmailPreferences();
+  const isEnabled = !emailPreferences.includes(EmailPreference.MARKETING_EMAILS);
+
+  const onEnable = () => {
+    const newPrefs = enableEmailPreference(EmailPreference.MARKETING_EMAILS, emailPreferences);
+    updateUserEmailPreferences({ variables: { emailPreferences: newPrefs } });
+  };
+  const onDisable = () => {
+    const newPrefs = disableEmailPreference(EmailPreference.MARKETING_EMAILS, emailPreferences);
+    updateUserEmailPreferences({ variables: { emailPreferences: newPrefs } });
```

Comment:
And if you make the signature changes and naming changes this line and `onEnable` get much more concise:

```
updateUserEmailPreferences(enable(EmailPreference.MARKETING_EMAILS, emailPreferences)
```